### PR TITLE
Sync v1 changes to v2: C*3.11 and C*4.0 upgrades.

### LINKED ISSUE #2384: Revert partial DSE update for SGv1 to get back to 6.8.29 (undo of DSE changes from #2359)
Update of DSE persistence to use 6.8.31 was incomplete in #2359, and cannot be completed quite yet (there is investigation on version compatibility). In the meantime we need to make sure `v1` can be built with consistent set of components, so let's make sure all references are to 6.8.29.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Merges v1 changes (update C*3.11 and C*4) to v2 (main)

**Which issue(s) this PR fixes**:
Fixes #2384 (part)

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 steps:
   - id: 'pull image'
-    name: 'gcr.io/stargateio/stargate-builder:v1.0.13'
+    name: 'gcr.io/stargateio/stargate-builder:v1.0.15'
```
Comment (ivansenic) [inline cloudbuild.yaml:3]: everywhere else is `v1.0.16`

File: pom.xml [hunk 2]
```diff
@@ -16,7 +16,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.scm.id>github</project.scm.id>
     <doclint>none</doclint>
-    <driver.version>4.14.1</driver.version>
+    <driver.version>4.15.0</driver.version>
```
Comment (ivansenic) [inline pom.xml:19]: where s this coming from?

**Review by ivansenic — COMMENTED** [review]
Is this a merge commit? What's the idea?

**Comment by tatu-at-datastax** [comment]
> Is this a merge commit? What's the idea?

Yes, merge commit.

File: pom.xml (hunk 2 above)
Comment (tatu-at-datastax) [reply pom.xml:19]: From v1. Updates other than DSE retained -- this included. Should not make a difference (I tried with both versions and DSE 6.8.30/.31 had same issue).

File: cloudbuild.yaml (hunk 1 above)
Comment (tatu-at-datastax) [reply cloudbuild.yaml:3]: Ah. thanks will fix.

**Review by jeffreyscarpenter — APPROVED** [review]

**Pull request merged**